### PR TITLE
Implemented collections deque

### PR DESCRIPTION
**Pull request by oneEdoubleD**
I've implemented the symbolic collections.deque class along with a few test cases and the change to `make_registrations()`. Let me know if there's anything else to do, or if you'd like any other test cases covered!

### CODE REVIEW

**Comment by oneEdoubleD** [comment]
This is related to https://github.com/pschanely/CrossHair/issues/3

File: crosshair/libimpl/collectionslib_test.py [hunk 1]
```diff
@@ -12,28 +13,102 @@
...
-            pre: len(l) > 0
-            post: len(l) != 222
-            '''
-            return l
-        # Right now, this returns unknown rather than fail, because creation of
-        # the deque forces the list length to become concrete.
-        self.assertEqual(*check_fail(f))
+
+    def setUp(self):
+        self.test_list = ListBasedDeque([1, 2, 3, 4, 5])
```
Comment (pschanely) [inline crosshair/libimpl/collectionslib_test.py]: Personal preference thing (so you should feel free to ignore if you like): I tend to bias towards repeating the test data in each test. It adds an extra line to a lot of tests, but makes them easier to read.

File: crosshair/libimpl/collectionslib.py [hunk 2]
```diff
@@ -8,14 +8,71 @@ class ListBasedDeque:
     def __init__(self, contents: List[T], maxlen: Optional[int]=None):
```
Comment (pschanely) [inline crosshair/libimpl/collectionslib.py]: Making the class extend `collections.abc.Sequence` is possibly a good idea?
Which makes me also think that some list-like deque behavior might not work properly yet?:
```
>>> from collections import deque
>>> 3 in deque([3,4])
True
>>> deque([3,4]) < deque([4,3])
True
>>> deque([3,4]) > deque([4,3])
False
>>> deque([3,4]) == deque([3,4])
True
```

File: crosshair/libimpl/collectionslib_test.py [hunk 3]
```diff
@@ -12,28 +13,102 @@
...
+            self.test_list.index(1, start=2)
+
+        self.assertTrue('1 is not in list' in str(context.exception))
+    
+    def test_deque_index_with_start_and_end_index(self) -> None:
+        i = self.test_list.index(2, start=0, end=3)
+        self.assertEqual(i, 1)
+    
+    def test_deque_index_with_start_and_end_index_throws_correct_exception(self) -> None:
+        with self.assertRaises(ValueError) as context:
```
Comment (pschanely) [inline crosshair/libimpl/collectionslib_test.py]: Thanks for exception checking!

**Comment by pschanely** [comment]
I'll merge what we've got so far. Let's continue to work on it, though!

**Pull request merged**